### PR TITLE
Create jpeg virtual package

### PR DESCRIPTION
**Pull request by jschueller**
like this @isuruf ?

### CODE REVIEW

**Comment by isuruf** [comment]
`libjpeg.8.dylib` from this package has `(compatibility version 8.0.0, current version 8.2.2)` while the one from `jpeg` package has `(compatibility version 13.0.0, current version 13.0.0)`. This has to be fixed before merging this.

**Comment by jschueller** [comment]
@isuruf ok

**Comment by isuruf** [comment]
Changing the soversion like this fixes the `current-version` and `comaptibility-version`, but breaks the install name.

**Comment by jakirkham** [comment]
How would you recommend we proceed then?

**Comment by ericpre** [comment]
Thanks @jschueller for tackling this! Since this seems to be a tricky, could one of the _conda-build/conda-forge wizards_ helps with some tricks?

This is one blocker for packaging imagecodecs: https://github.com/conda-forge/staged-recipes/pull/10331.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -22,11 +23,41 @@ requirements:
...
-test:
-  files:
-    - testorig.jpg
-  commands:
-    - djpeg -dct int -ppm -outfile testout.ppm testorig.jpg
+outputs:
+  - name: libjpeg-turbo
+    build:
+      run_exports:
+      - {{ pin_subpackage('libjpeg-turbo', max_pin='x') }}
```
Comment (hmaarrfk) [inline recipe/meta.yaml:30]:
```suggestion
      - {{ pin_subpackage('libjpeg-turbo', exact=true) }}
```

File: recipe/meta.yaml [hunk 2]
```diff
@@ -22,11 +23,41 @@ requirements:
...
+  - name: jpeg
+    version: 8d
+    build:
+      run_exports:
+        - {{ pin_subpackage('jpeg') }}
+      track_features:
+        - jpeg_impl_libjpeg_turbo
+    requirements:
+      run:
+        - libjpeg-turbo
```
Comment (hmaarrfk) [inline recipe/meta.yaml:58]:
```suggestion
        - {{ pin_subpackage('libjpeg-turbo', exact=True) }}
```

**Comment by hmaarrfk** [comment]
> Changing the soversion like this fixes the current-version and comaptibility-version, but breaks the install name.

Could we mark all the old builds as broken? does anything actually use `jpeg-turbo` as of today?

**Comment by jschueller** [comment]
@conda-forge-admin please rerender

**Comment by jschueller** [comment]
@isuruf I fixed the compatibility/current version property, I believe this is ready for review

**Comment by isuruf** [comment]
Let's continue the discussion at https://github.com/conda-forge/conda-forge.github.io/issues/673

**Comment by jschueller** [comment]
@conda-forge-admin please rerender

**Comment by hmaarrfk** [comment]
i think we can close this.

Thank you for helping get to this stage!

**Pull request closed without merging**